### PR TITLE
Process commands

### PR DESCRIPTION
**Pull request by jnicklas**
This is working towards actually running the tests. Once the command server receives a run mutation it will enqueue this in the orchestrator which will then pick it up and forward it to the agent. The agent will then proceed to run the supplied tests. We're handing the agent a subset of the manifest (or the entire manifest), which it will recursively descend through.

### CODE REVIEW

File: packages/agent/app/agent.ts [hunk 1]
```diff
@@ -1,59 +1,44 @@
...
-      scriptElement.src = message.manifest;
-      scriptElement.addEventListener('load', () => {
-        console.log('[agent] loaded test manifest', __bigtestManifest);
-      });
-      document.body.appendChild(scriptElement);
+  while(true) {
+    let message = yield connection.receive();
 
+    if(message.type === "run") {
+      fork(run(testFrame, message));
```
Comment (cowboyd) [inline packages/agent/app/agent.ts]: Looks like the `fork` expression won't ever run.

File: packages/agent/app/agent.ts [hunk 2]
```diff
@@ -1,59 +1,44 @@
...
-  console.log('[agent] connecting to orchestrator at', connectTo);
-  let socket = new WebSocket(connectTo);
-
-  socket.addEventListener('open', () => {
-    socket.send(JSON.stringify({
-      type: 'connected',
-      data: Bowser.parse(navigator.userAgent)
-    }));
-    console.log('[agent] socket connection established');
+  fork(function*() {
```
Comment (cowboyd) [inline packages/agent/app/agent.ts]: Here's another "lone" fork?

File: packages/agent/app/index.html [hunk 3]
```diff
@@ -19,6 +42,9 @@
   </head>
 
   <body>
+    <div id="splash">
+      <img src="./assets/bigtest-logo.svg">
```
Comment (cowboyd) [inline packages/agent/app/index.html]: 😍 😍 😍 

If the agent is going to just be standing around, might as well be wearing nice clothes, right? love to see it...

File: packages/agent/app/socket-connection.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { Operation } from 'effection';
+import { Mailbox } from './effection/mailbox';
+
+export class SocketConnection {
+  private connection: WebSocket;
+
+  static *start(connectTo: string) {
+    let socket = new WebSocket(connectTo);
+    let mailbox = yield Mailbox.watch(socket, ['open', 'message', 'close']);
```
Comment (cowboyd) [inline packages/agent/app/socket-connection.ts]: Should this mailbox be watching in the parent scope?

**Comment by jnicklas** [comment]
@cowboyd one thing in this PR is that it basically copy-pastes the mailbox implementation into this package, that's not ideal, of course. Can we extract this into its own package, maybe? My packaging-fu is not strong enough to work this out myself.

**Comment by cowboyd** [comment]
Should we make it `@bigtest/events`? or go through the ordeal of setting up `effection` as a monerepo. Alternatively, we could actually set up `@effection/events` inside the bigtest monorepo since @frontsidejack is the one with the credentials to publish both of those NPM packages.

**Comment by jnicklas** [comment]
This has now all been extracted to other PRs, the last bit that hasn't been merged yet is in #205

**Pull request closed without merging**